### PR TITLE
[nae_prechange] Add tasks to nae_prechange and add better error message. fixes #2, #42 and #43

### LINKED ISSUE #2: Better error message for unparsed input file when run without verify:True (nae_prechange)
Ansible NAE  is unable to parse the JSON configuration created by the ansible-aci module version 0.0.7

Steps to reproduce:

1. Install the latest NAE and ACI Module
2. Run the [full_aci_demo playbook](https://github.com/CiscoDevNet/ansible-nae/blob/master/playbooks/full_aci_demo.yaml ) to generate the ACI config file 
3. NAE module fails with Couldn't parse change file message

### LINKED ISSUE #42: Documentation Issue: nae_prechange
Hi, 

seems like I found an Issue in the documentation of the nae_prechange module.

image

The query NAE prechange explanation is in there twice and on the first occurrence it is stated that the state should be "verify"

### LINKED ISSUE #43: nae_prechange fails on query state
Hi, 

I got a working playbook to create and query the nae prechange analysis. 
The query is working until the pre change analysis is done and when it gets the status "completed" the query is failing with following module error.

"Traceback (most recent call last):\n  File \"/var/lib/awx/.ansible/tmp/ansible-tmp-1607952875.8573136-10584-170220307643012/AnsiballZ_nae_prechange.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/var/lib/awx/.ansible/tmp/ansible-tmp-1607952875.8573136-10584-170220307643012/AnsiballZ_nae_prechange.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/var/lib/awx/.ansible/tmp/ansible-tmp-1607952875.8573136-10584-170220307643012/AnsiballZ_nae_prechange.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.cisco.nae.plugins.modules.nae_prechange', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_cisco.nae.nae_prechange_payload_88bf6y31/ansible_cisco.nae.nae_prechange_payload.zip/ansible_collections/cisco/nae/plugins/modules/nae_prechange.py\", line 215, in <module>\n  File \"/tmp/ansible_cisco.nae.nae_prechange_payload_88bf6y31/ansible_cisco.nae.nae_prechange_payload.zip/ansible_collections/cisco/nae/plugins/modules/nae_prechange.py\", line 198, in main\n  File \"/tmp/ansible_cisco.nae.nae_prechange_payload_88bf6y31/ansible_cisco.nae.nae_prechange_payload.zip/ansible_collections/cisco/nae/plugins/module_utils/nae.py\", line 469, in get_pre_change_result\nTypeError: argument of type 'NoneType' is not iterable\n",

The playbook looks as following.

` - name: Play 1 Check Epoch Delta
  hosts: all
  connection: local

  tasks:
  
    - name: Query Prechange analysis
      cisco.nae.nae_prechange:
        host: "{{ nae }}"
        port: 443
        username: "{{ lookup('env', 'ANSIBLE_NET_USERNAME') }}"
        password: "{{ lookup('env', 'ANSIBLE_NET_PASSWORD') }}"
        ag_name: "{{ agName }}"
        name: NAEAnalysis
        state: query
      delegate_to: localhost
`

Please advice if I am holding it wrong :)

### PR DESCRIPTION
**Pull request by anvitha-jain**
Resolves issues #2 #43 #42  by adding tasks to nae_prechange - main.yml and added condition to check epoch value in nae.py and made changes to support the prechange tasks.

### CODE REVIEW

File: plugins/module_utils/nae.py [hunk 1]
```diff
@@ -578,7 +582,8 @@ def send_manual_payload(self):
             self.result['Result'] = "Pre-change analysis %(name)s successfully created." % self.params
 
         elif '5.0' in self.version or '5.1' in self.version:
-            url = 'https://%(host)s:%(port)s/nae/api/v1/config-services/prechange-analysis/manual-changes?action=RUN' % self.params
+            # job_id = self.params.get('job_id') # necessary to PUT method
```
Comment (lhercot) [inline plugins/module_utils/nae.py]: If not need, we should remove it

File: tests/integration/targets/nae_prechange/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                                "pcv_status": ""
+                            }
+                        }
+                    }]
+                }
+            }
+        ]
+    name: TestAnalysis_2
+    state: present
+    # save: True
```
Comment (lhercot) [inline tests/integration/targets/nae_prechange/tasks/main.yml]: Title say save but this line is not

File: tests/integration/targets/nae_prechange/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                    }
+                }
+            }
+        ]
+    name: New_Analysis
+    state: present
+  register: add_pre_change
+  ignore_errors: true
+
+- name: Verify Addding pre-change analysis from manual changes
```
Comment (lhercot) [inline tests/integration/targets/nae_prechange/tasks/main.yml]: Addding -> adding

File: tests/integration/targets/nae_prechange/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+        ]
+    name: TestAnalysis
+    state: present
+  delegate_to: localhost
+  ignore_errors: yes
+  register: add_pre_change_again
+
+- name: Verify Addding pre-change analysis from manual changes
+  assert:
+    that:
```
Comment (lhercot) [inline tests/integration/targets/nae_prechange/tasks/main.yml]: is it changed again or not?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: tests/integration/targets/nae_prechange/tasks/main.yml (hunk 4 above)
Comment (anvitha-jain) [reply tests/integration/targets/nae_prechange/tasks/main.yml]: no

File: plugins/module_utils/nae.py [hunk 5]
```diff
@@ -350,7 +350,9 @@ def get_pre_change_analyses(self):
 
         if resp.headers.get('Content-Encoding') == "gzip":
             r = gzip.decompress(resp.read())
+            # self.result['stdout'] = json.loads(r.decode())['value']['data']
```
Comment (lhercot) [inline plugins/module_utils/nae.py]: lets remove those comments if we don't need them

File: plugins/module_utils/nae.py [hunk 6]
```diff
@@ -578,73 +584,141 @@ def send_manual_payload(self):
...
+            obj = self.get_pre_change_analysis()
+            if obj is not None:
+                self.params['job_id'] = obj.get('job_id')
+                if not self.issubset(form, obj):
+                    if obj.get('analysis_status') != 'SAVED':
+                        self.module.fail_json(
+                            msg='Pre-change analysis {0} is not in SAVED status. It cannot be edited.'.format(self.params.get('name')),
+                            **self.result)
+                    else:
+                        # self.params['job_id'] = obj.get('job_id')
```
Comment (lhercot) [inline plugins/module_utils/nae.py]: Remove comment if not needed

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**